### PR TITLE
Feat setting object

### PR DESCRIPTION
**Pull request by AustinJMann**

### CODE REVIEW

File: src/core/SettingsManager.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    static {
+        SettingsManager.reset()
+        const settingsJson = localStorage.getItem('settings');
+        if (settingsJson) SettingsManager.settings = JSON.parse(settingsJson);
+    }
+
+    public static write(): void {
+        localStorage.setItem('settings', JSON.stringify(SettingsManager.settings));
+        console.log(localStorage.getItem('settings'));
```
Comment (IsaacThoman) [inline src/core/SettingsManager.ts:12]: remove the console.log pls

File: src/core/SettingsManager.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+export class SettingsManager {
+    static settings: { [key: string]: string | number | boolean | null};
```
Comment (IsaacThoman) [inline src/core/SettingsManager.ts:2]: i'd probably just manually name every setting here for existence/type clarity

**Review by IsaacThoman — CHANGES_REQUESTED** [review]
looks nice, address above comments pls :)

File: src/core/SettingsManager.ts (hunk 1 above)
Comment (AustinJMann) [reply src/core/SettingsManager.ts:12]: lol oops

**Review by IsaacThoman — APPROVED** [review]

**Pull request merged**